### PR TITLE
Catch unhandled exceptions

### PR DESCRIPTION
**Pull request by graetzer**
### Scope & Purpose

Catch unhandled protocl exceptions

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)

### Testing & Verification

This change is already covered by existing tests

http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6245/

### CODE REVIEW

File: arangod/GeneralServer/GeneralCommTask.cpp [hunk 1]
```diff
@@ -75,17 +75,24 @@ void GeneralCommTask<T>::asyncReadSome() {
...
         break;
       }
-      if (!readCallback(ec)) {
-        return;
+      
+      try {
+        cont = readCallback(ec);
+        available = _protocol->available(ec);
+      } catch (...) {
+        LOG_TOPIC("2c6b5", ERR, arangodb::Logger::REQUESTS)
```
Comment (dothebart) [inline arangod/GeneralServer/GeneralCommTask.cpp]: I feel that this could also justify `FATAL` - since callbacks shouldn't let exceptions out?

Comment (graetzer) [reply arangod/GeneralServer/GeneralCommTask.cpp]: but it is not fatal

**Comment by graetzer** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/6247/

File: arangod/GeneralServer/GeneralCommTask.cpp (hunk 1 above)
Comment (dothebart) [reply arangod/GeneralServer/GeneralCommTask.cpp]: if maintainer mode?

**Pull request merged**